airflow/models/renderedtifields.py

```diff
@@ -57,9 +57,20 @@ def __init__(self, ti: TaskInstance, render_templates=True):
             field: serialize_template_field(getattr(self.task, field)) for field in self.task.template_fields
         }
 
+        self._redact()
+
     def __repr__(self):
         return f"<{self.__class__.__name__}: {self.dag_id}.{self.task_id} {self.execution_date}"
 
+    def _redact(self):
+        from airflow.utils.log.secrets_masker import redact
+
+        if self.k8s_pod_yaml:
+            self.k8s_pod_yaml = redact
```

Comment:
Is this correct ? I am not sure what's happening here? It is a dict originally and we are replacing it with `redact` method?

Reply:
No, this is not correct at all, which means it isn't covered by tests!

I'll expand this

Reply:
Oh, no, is already covered -- tests are failing cos of it